File: src/main/java/me/cephetir/skyskipped/features/impl/HidePetCandies.java
/*
 *
 * DO WHAT THE FUCK YOU WANT TO PUBLIC LICENSE
 * Version 2, December 2004
 *
 * Copyright (C) 2021 Cephetir
 *
 * Everyone is permitted to copy and distribute verbatim or modified
 * copies of this license document, and changing it is allowed as long
 * as the name is changed.
 *
 * DO WHAT THE FUCK YOU WANT TO PUBLIC LICENSE
 * TERMS AND CONDITIONS FOR COPYING, DISTRIBUTION AND MODIFICATION
 *
 *  0. You just DO WHAT THE FUCK YOU WANT TO.
 */

package me.cephetir.skyskipped.features.impl;

import me.cephetir.skyskipped.config.Config;
import me.cephetir.skyskipped.features.Feature;
import net.minecraftforge.event.entity.player.ItemTooltipEvent;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;

public class HidePetCandies extends Feature {

    public HidePetCandies() {
        super("HidePetCandies", "Visual", "Hide pet's candies counter in tooltip.");
    }

    @SubscribeEvent
    public void onTooltip(ItemTooltipEvent event) {
        if (!Config.hidePetCandies) return;
        for (int i = 0; i < event.toolTip.size(); i++) {
            if (event.toolTip.get(i).contains("Pet Candy Used")) {
                event.toolTip.remove(i);
                event.toolTip.remove(i - 1);
            }
        }
    }
}

File: src/main/kotlin/me/cephetir/skyskipped/features/impl/visual/HidePetCandies.kt
/*
 * SkySkipped - Hypixel Skyblock QOL mod
 * Copyright (C) 2023  Cephetir
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published
 * by the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package me.cephetir.skyskipped.features.impl.visual

import me.cephetir.skyskipped.config.Config
import me.cephetir.skyskipped.features.Feature
import net.minecraftforge.event.entity.player.ItemTooltipEvent
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent

class HidePetCandies : Feature() {

    @SubscribeEvent
    fun onTooltip(event: ItemTooltipEvent) {
        if (!Config.hidePetCandies.value) return
        for (i in 0 until event.toolTip.size - 1) {
            if (event.toolTip[i].contains("Pet Candy Used")) {
                event.toolTip.removeAt(i)
                event.toolTip.removeAt(i - 1)
                break
            }
        }
    }
}